internal/controller/controller.go

```diff
@@ -78,13 +78,17 @@ func (c *Controller) Start(ctx context.Context) error {
 	// OTel semantic introduced in https://github.com/open-telemetry/semantic-conventions/issues/66
 	metadataCollector.AddCustomData("os.kernel.release", kernelVersion)
 
-	// hostname and sourceIP will be populated from the root namespace.
-	hostname, sourceIP, err := helpers.GetHostnameAndSourceIP(c.config.CollAgentAddr)
-	if err != nil {
-		log.Warnf("Failed to fetch metadata information in the root namespace: %v", err)
+	if c.config.CollAgentAddr != "" {
+		// hostname and sourceIP will be populated from the root namespace.
+		hostname, sourceIP,
+			err := //nolint:govet // shadowing err is fine, previous errors have been handled
+			helpers.GetHostnameAndSourceIP(c.config.CollAgentAddr)
+		if err != nil {
+			log.Warnf("Failed to fetch metadata information in the root namespace: %v", err)
+		}
+		metadataCollector.AddCustomData("host.name", hostname)
+		metadataCollector.AddCustomData("host.ip", sourceIP)
```

Comment:
We're already calling `helpers.GetHostnameAndSourceIP` in [main](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/aea5a1717f2d5487aca5f07bd0422d53412eed55/main.go#L111-L135) once so we shouldn't call it again here, but instead allow for `hostname` and `sourceIP` to be set in the `controller.Config` after the first [call](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/aea5a1717f2d5487aca5f07bd0422d53412eed55/main.go#L112) in `main.go`.

The logic here then would become:

```suggestion
	metadataCollector.AddCustomData("host.name", c.config.HostName)
	metadataCollector.AddCustomData("host.ip", c.config.IPAddress)
```

This allows for the entity that sets up the controller to pass in these values.